### PR TITLE
Label physicians and churches

### PR DESCRIPTION
**Pull request by az0**

### CODE REVIEW

**Comment by az0** [comment]
@jeancochrane : 
Here I made the changes you requested [in PR 55](https://github.com/datamade/probablepeople/pull/55#issuecomment-337723405)

File: name_data/labeled/company_labeled.xml [hunk 1]
```diff
@@ -1421,4 +1421,57 @@
...
+  <Name><CorporationName>Abyssinian</CorporationName> <CorporationNameOrganization>Baptist</CorporationNameOrganization> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
+  <Name><CorporationName>Urban</CorporationName> <CorporationName>Grace</CorporationName> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
+  <Name><CorporationName>Temple</CorporationName> <CorporationName>B'Nai</CorporationName> <CorporationName>Abraham</CorporationName></Name>
+  <Name><CorporationName>Madison</CorporationName> <CorporationName>Avenue</CorporationName> <CorporationNameOrganization>Baptist</CorporationNameOrganization> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
+  <Name><CorporationName>St.</CorporationName> <CorporationName>Joseph</CorporationName> <CorporationNameOrganization>Basilica</CorporationNameOrganization></Name>
+  <Name><CorporationName>Trinity</CorporationName> <CorporationName>Evangelical</CorporationName> <CorporationNameOrganization>Lutheran</CorporationNameOrganization> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
```
Comment (jeancochrane) [inline name_data/labeled/company_labeled.xml]: `Evangelical` is likely a `CorporationNameOrganization` here, no? Seems like it's a modifier for `Lutheran`, as opposed to `Trinity`.

File: name_data/labeled/company_labeled.xml [hunk 2]
```diff
@@ -1421,4 +1421,57 @@
...
+  <Name><CorporationName>Jubilee</CorporationName> <CorporationName>Christian</CorporationName> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
```
Comment (jeancochrane) [inline name_data/labeled/company_labeled.xml]: Ditto for `Christian`.

File: name_data/labeled/company_labeled.xml [hunk 3]
```diff
@@ -1421,4 +1421,57 @@
...
+  <Name><CorporationName>Saint</CorporationName> <CorporationName>Vincent</CorporationName> <CorporationName>de</CorporationName> <CorporationName>Paul</CorporationName></Name>
+  <Name><CorporationName>Duke</CorporationName> <CorporationNameOrganization>Chapel</CorporationNameOrganization></Name>
+  <Name><CorporationName>St.</CorporationName> <CorporationName>Augustine</CorporationName> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
+  <Name><CorporationName>Beulah</CorporationName> <CorporationNameOrganization>Baptist</CorporationNameOrganization> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
+  <Name><CorporationName>Hopeful</CorporationName> <CorporationNameOrganization>Lutheran</CorporationNameOrganization> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
+  <Name><CorporationName>Salem</CorporationName> <CorporationName>Evangelical</CorporationName> <CorporationNameOrganization>Lutheran</CorporationNameOrganization> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
```
Comment (jeancochrane) [inline name_data/labeled/company_labeled.xml]: Similar question about `Evangelical` here.

File: name_data/labeled/company_labeled.xml [hunk 4]
```diff
@@ -1421,4 +1421,57 @@
...
+  <Name><CorporationName>Rehoboth</CorporationName> <CorporationNameOrganization>Methodist</CorporationNameOrganization> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
+  <Name><CorporationName>San</CorporationName> <CorporationName>Ignacio</CorporationName> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
+  <Name><CorporationName>Robeline</CorporationName> <CorporationNameOrganization>Methodist</CorporationNameOrganization> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
+  <Name><CorporationName>Taveau</CorporationName> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
+  <Name><CorporationName>Brady</CorporationName> <CorporationName>Memorial</CorporationName> <CorporationNameOrganization>Chapel</CorporationNameOrganization></Name>
+  <Name><CorporationName>Zion</CorporationName> <CorporationName>Evangelical</CorporationName> <CorporationNameOrganization>Lutheran</CorporationNameOrganization> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
```
Comment (jeancochrane) [inline name_data/labeled/company_labeled.xml]: Another instance of `Evangelical`.

File: name_data/labeled/person_labeled.xml [hunk 5]
```diff
@@ -3995,4 +3995,93 @@
...
   <Name><GivenName>Franco</GivenName> <Surname>Coladipietro</Surname></Name>
   <Name><GivenName>Carol</GivenName> <MiddleName>Sente</MiddleName></Name>
+  <Name><PrefixMarital>Ms</PrefixMarital> <GivenName>RESHMA</GivenName> <Surname>STAFFORD</Surname></Name>
+  <Name><PrefixMarital>Miss</PrefixMarital> <GivenName>ADRINE</GivenName> <MiddleName>DJ</MiddleName> <Surname>VARTANI</Surname></Name>
+  <Name><PrefixMarital>Miss</PrefixMarital> <GivenName>SOPHIA</GivenName> <MiddleName>SHIOW-HUEY</MiddleName> <Surname>WU</Surname></Name>
+  <Name><PrefixMarital>Ms</PrefixMarital> <GivenName>NNEMDI</GivenName> <MiddleInitial>A</MiddleInitial> <Surname>BAIRD</Surname></Name>
+  <Name><PrefixMarital>Ms.</PrefixMarital> <GivenName>MA</GivenName> <GivenName>VICTORIA</GivenName> <MiddleInitial>C</MiddleInitial> <Surname>DELA</Surname> <Surname>CRUZ</Surname></Name>
+  <Name><PrefixMarital>Ms</PrefixMarital> <GivenName>SWATI</GivenName> <Surname>DHANKIKAR</Surname></Name>
+  <Name><PrefixMarital>Mr</PrefixMarital> <GivenName>LEWIS</GivenName> <MiddleInitial>E</MiddleInitial> <Surname>SELLERS</Surname> <SuffixOther>III</SuffixOther></Name>
```
Comment (jeancochrane) [inline name_data/labeled/person_labeled.xml]: I think `III` should be `SuffixGenerational` in this case.

File: name_data/labeled/person_labeled.xml [hunk 6]
```diff
@@ -3995,4 +3995,93 @@
...
+  <Name><PrefixOther>Dr.</PrefixOther> <GivenName>THRESSIAMMA</GivenName> <Surname>CHOLANKERIL</Surname> <SuffixOther>M.D.</SuffixOther></Name>
+  <Name><PrefixMarital>Miss</PrefixMarital> <GivenName>LEESA</GivenName> <MiddleName>ANN</MiddleName> <Surname>VITEK</Surname></Name>
+  <Name><PrefixMarital>Miss</PrefixMarital> <GivenName>EILEEN</GivenName> <Surname>JANEC,</Surname> <SuffixOther>MD</SuffixOther></Name>
+  <Name><PrefixMarital>Mr.</PrefixMarital> <And>&amp;</And> <PrefixMarital>Mrs.</PrefixMarital> <GivenName>EN</GivenName> <GivenName>SU</GivenName> <Surname>LIN</Surname> <SuffixOther>M.D.</SuffixOther></Name>
+  <Name><PrefixOther>Dr</PrefixOther> <And>&amp;</And> <PrefixMarital>Mrs</PrefixMarital> <GivenName>DERIC</GivenName> <MiddleInitial>C</MiddleInitial> <Surname>MCINTOSH</Surname></Name>
+  <Name><PrefixMarital>Miss</PrefixMarital> <GivenName>KELLY</GivenName> <MiddleName>MARIE</MiddleName> <Surname>JENSEN</Surname></Name>
+  <Name><GivenName>TOAN</GivenName> <MiddleName>QUOC</MiddleName> <Surname>TRAN</Surname></Name>
+  <Name><PrefixMarital>Miss</PrefixMarital> <GivenName>KATHLEEN</GivenName> <Surname>KIRCHMEIER</Surname> <Surname>KROESSLER</Surname></Name>
```
Comment (jeancochrane) [inline name_data/labeled/person_labeled.xml]: Should `KIRCHMEIER` be a `MiddleName` in this case?

File: name_data/labeled/person_labeled.xml [hunk 7]
```diff
@@ -3995,4 +3995,93 @@
...
+  <Name><PrefixMarital>Miss</PrefixMarital> <GivenName>URVI</GivenName> <MiddleName>PRADEEP</MiddleName> <Surname>VYAS</Surname></Name>
+  <Name><PrefixMarital>Ms</PrefixMarital> <GivenName>NEVEDITA</GivenName> <MiddleInitial>S</MiddleInitial> <Surname>SARASWAT</Surname> <SuffixOther>MD</SuffixOther></Name>
+  <Name><PrefixMarital>Miss</PrefixMarital> <GivenName>JEWELL</GivenName> <MiddleName>ANN</MiddleName> <Surname>CRAWFORD</Surname></Name>
+  <Name><PrefixOther>Dr</PrefixOther> <And>&amp;</And> <PrefixMarital>Mrs</PrefixMarital> <GivenName>RUSKIN</GivenName> <Surname>LINES</Surname></Name>
+  <Name><PrefixMarital>Miss</PrefixMarital> <GivenName>JENNA</GivenName> <Surname>LAMMERS</Surname> <Surname>MCCREE</Surname></Name>
+  <Name><GivenName>HINA</GivenName> <Surname>AAMIR</Surname></Name>
+  <Name><PrefixMarital>Miss</PrefixMarital> <GivenName>MARIETTA</GivenName> <MiddleInitial>F</MiddleInitial> <Surname>ABALOS</Surname> <Surname>GALITO</Surname></Name>
+  <Name><PrefixMarital>Miss</PrefixMarital> <GivenName>CHRISTINA</GivenName> <Surname>MERCY</Surname> <Surname>KEMP</Surname></Name>
```
Comment (jeancochrane) [inline name_data/labeled/person_labeled.xml]: Similar question about whether `MERCY` should be a `MiddleName`.

File: name_data/labeled/person_labeled.xml [hunk 8]
```diff
@@ -3995,4 +3995,93 @@
...
+  <Name><PrefixMarital>Miss</PrefixMarital> <GivenName>TRACEY</GivenName> <MiddleName>MARIE</MiddleName> <MiddleName>MIRANDA</MiddleName> <Surname>RACO</Surname></Name>
+  <Name><GivenName>EDWARD</GivenName> <MiddleName>JAY</MiddleName> <Surname>POLLYEA</Surname></Name>
+  <Name><PrefixOther>Dr.</PrefixOther> <And>and</And> <PrefixMarital>Mrs.</PrefixMarital> <FirstInitial>E</FirstInitial> <GivenName>WILLIAM</GivenName> <MiddleInitial>W</MiddleInitial> <Surname>JOHNSON</Surname></Name>
+  <Name><PrefixMarital>Mr.</PrefixMarital> <GivenName>A</GivenName> <GivenName>Y</GivenName> <GivenName>M</GivenName> <GivenName>MAKSUDUL</GivenName> <Surname>CHOWDHURY</Surname></Name>
+  <Name><GivenName>CHALERMCHAI</GivenName> <Surname>PUNYA</Surname></Name>
+  <Name><GivenName>BING</GivenName> <MiddleName>WANG</MiddleName> <Surname>GALE</Surname></Name>
+  <Name><GivenName>ALI</GivenName> <MiddleInitial>A</MiddleInitial> <Surname>AL</Surname> <Surname>MUDAMGHA</Surname></Name>
+  <Name><PrefixMarital>Miss</PrefixMarital> <GivenName>DYANN</GivenName> <Surname>HOLLOWAY</Surname> <Surname>DALEY</Surname></Name>
```
Comment (jeancochrane) [inline name_data/labeled/person_labeled.xml]: Is `HOLLOWAY` a `MiddleName` as well?

File: name_data/labeled/person_labeled.xml [hunk 9]
```diff
@@ -3995,4 +3995,93 @@
...
+  <Name><PrefixMarital>Miss</PrefixMarital> <GivenName>JENNIFER</GivenName> <Surname>WISDOM</Surname> <Surname>BEHOUNEK</Surname></Name>
```
Comment (jeancochrane) [inline name_data/labeled/person_labeled.xml]: Similar question about `WISDOM` being a `MiddleName`.

File: name_data/labeled/person_labeled.xml [hunk 10]
```diff
@@ -3995,4 +3995,93 @@
...
+  <Name><GivenName>SULAGNA</GivenName> <Surname>MOOKHERJEE</Surname></Name>
+  <Name><GivenName>AKI</GivenName> <Surname>INOUE</Surname></Name>
+  <Name><GivenName>EL</GivenName> <GivenName>AGATHE</GivenName> <MiddleInitial>A</MiddleInitial> <Surname>TEMBENG</Surname></Name>
+  <Name><GivenName>ESMIRALDA</GivenName> <Surname>YEREMEYEVA</Surname> <Surname>HENDERSON</Surname></Name>
+  <Name><PrefixMarital>Mr</PrefixMarital> <GivenName>VERRAD</GivenName> <MiddleName>KWASI</MiddleName> <MiddleName>OWUSU</MiddleName> <Surname>NYAME</Surname></Name>
+  <Name><GivenName>MONTGOMERY</GivenName> <Surname>TIMMS</Surname></Name>
+  <Name><GivenName>TEMEKIS</GivenName> <MiddleName>DELVAUGHN</MiddleName> <Surname>HAMPTON</Surname></Name>
+  <Name><GivenName>HANNAH</GivenName> <Surname>PEAGLER</Surname> <Surname>HOWARD</Surname></Name>
```
Comment (jeancochrane) [inline name_data/labeled/person_labeled.xml]: `PEAGLER` as a `MiddleName`?

File: name_data/labeled/person_labeled.xml [hunk 11]
```diff
@@ -3995,4 +3995,93 @@
...
+  <Name><PrefixMarital>Miss</PrefixMarital> <GivenName>LUCIA</GivenName> <Surname>FONTES</Surname> <Surname>BORTS</Surname></Name>
```
Comment (jeancochrane) [inline name_data/labeled/person_labeled.xml]: `FONTES` as a `MiddleName`?

File: tests/test_data_labeled.xml [hunk 12]
```diff
@@ -65,4 +65,14 @@
   <Name><CorporationName>ENGLEWOOD</CorporationName> <CorporationName>MARKET,</CorporationName> <CorporationLegalType>INC.</CorporationLegalType></Name>
   <Name><CorporationName>GRABTAXI</CorporationName> <CorporationName>HOLDINGS,</CorporationName> <CorporationLegalType>PTE</CorporationLegalType> <CorporationLegalType>LTD</CorporationLegalType></Name>
   <Name><CorporationName>VIKI</CorporationName> <CorporationName>ENGINEERING</CorporationName> <CorporationLegalType>PTE.</CorporationLegalType> <CorporationLegalType>LTD.</CorporationLegalType></Name>
+  <Name><PrefixMarital>Miss</PrefixMarital> <GivenName>LARA</GivenName> <MiddleInitial>E</MiddleInitial> <Surname>STROTHEIDE</Surname></Name>
```
Comment (jeancochrane) [inline tests/test_data_labeled.xml]: In general we try not to use the same samples in the training and testing sets. Do you mind removing the corresponding samples from the training data for these tests?

File: name_data/labeled/company_labeled.xml [hunk 13]
```diff
@@ -1421,4 +1421,57 @@
...
   <Name><CorporationName>Spacemob</CorporationName> <CorporationLegalType>Pte.</CorporationLegalType> <CorporationLegalType>Ltd.</CorporationLegalType></Name>
   <Name><CorporationName>NTUC</CorporationName> <CorporationName>LearningHub</CorporationName> <CorporationLegalType>Pte</CorporationLegalType> <CorporationLegalType>Ltd</CorporationLegalType></Name>
+  <Name><CorporationName>Bethlehem</CorporationName> <CorporationNameOrganization>Lutheran</CorporationNameOrganization> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
+  <Name><CorporationName>Salina</CorporationName> <CorporationName>Presbyterian</CorporationName> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
+  <Name><CorporationName>Ebenezer</CorporationName> <CorporationNameOrganization>Lutheran</CorporationNameOrganization> <CorporationNameOrganization>Chapel</CorporationNameOrganization></Name>
+  <Name><CorporationName>Brayton</CorporationName> <CorporationNameOrganization>Methodist</CorporationNameOrganization> <CorporationName>Episcopal</CorporationName> <CorporationNameOrganization>Church</CorporationNameOrganization></Name>
```
[the remark this answers is not in the document]
Comment (az0) [reply name_data/labeled/company_labeled.xml]: Yes, I agree

File: name_data/labeled/company_labeled.xml (hunk 1 above)
Comment (az0) [reply name_data/labeled/company_labeled.xml]: Yes, I agree

File: name_data/labeled/person_labeled.xml (hunk 5 above)
Comment (az0) [reply name_data/labeled/person_labeled.xml]: Good catch

File: name_data/labeled/person_labeled.xml (hunk 6 above)
Comment (az0) [reply name_data/labeled/person_labeled.xml]: According to the original CMS Physicians Data set for this individual, her middle_name is blank and last_name="KIRCHMEIER KROESSLER".

Also according to a glance at Google search results, KIRCHMEIER seems like a last name (like Smith, unlike Kathleen).

File: name_data/labeled/person_labeled.xml (hunk 7 above)
Comment (az0) [reply name_data/labeled/person_labeled.xml]: Like the other individuals, this came from the CMS Physicians Data Set. There the values are
first_name=Christina
middle_name=(Blank)
last_name=MERCY KEMP

Maybe her maiden name was Christina Mercy, and after marring Mr. Kemp, Christina changed her name to Christina Mercy Kemp. In this data set other female surnames (and some male males) are hyphenated, so maybe Christina never hyphenates her name or the hyphen was omitted accidentally in this record.

File: name_data/labeled/person_labeled.xml (hunk 8 above)
Comment (az0) [reply name_data/labeled/person_labeled.xml]: Yes, according to the original CMS Physicians data set HOLLOWAY here is a middle name. However, it looks more like a last name, so it seems the original data set inconsistently records the names of females with two last names written without a hyphen.

I will change this in the XML

File: name_data/labeled/person_labeled.xml (hunk 9 above)
Comment (az0) [reply name_data/labeled/person_labeled.xml]: No, in the original CMS Physicians data set middle_name is blank and last_name="WISDOM BEHOUNEK"

File: name_data/labeled/person_labeled.xml (hunk 10 above)
Comment (az0) [reply name_data/labeled/person_labeled.xml]: Yes, in the original data set this is the middle name. Like the other cases, this one looks like a last name, and it seems the original data set is inconsistent.

I will change it in the XML

File: name_data/labeled/person_labeled.xml (hunk 11 above)
Comment (az0) [reply name_data/labeled/person_labeled.xml]: No, in the original data set the last_name="FONTES BORTS" and middle_name is blank

File: tests/test_data_labeled.xml (hunk 12 above)
Comment (az0) [reply tests/test_data_labeled.xml]: Yes, I will

**Comment by az0** [comment]
I replied to your review comments about individuals, but lately I have not had much time to work on this project, so I may not respond soon to everything else

**Comment by fgregg** [comment]
@jeancochrane, where are we with this?

**Comment by jeancochrane** [comment]
@fgregg still waiting on @az0 to push changes based on my comments. Any updates on that?

**Comment by az0** [comment]
In some cases I need to make changes, and in other cases it seems the original was OK.

I have not had time to work on this.

File: name_data/labeled/company_labeled.xml (hunk 2 above)
Comment (az0) [reply name_data/labeled/company_labeled.xml]: I agree

File: name_data/labeled/company_labeled.xml (hunk 3 above)
Comment (az0) [reply name_data/labeled/company_labeled.xml]: I agree. I will change it.

**Comment by az0** [comment]
@jeancochrane 

For each item you raised I either made a code change or provided an explanation. Please review again. Thank you for your patience.

**Comment by az0** [comment]
@jeancochrane  The Travis CI checks failed on test performance. Do you recommend adding more training examples (which are not identical to the test set) until the tests pass?

**Comment by az0** [comment]
@jeancochrane Any thoughts on how to do with the test performance? Or any advice on how to get this merged?

**Comment by az0** [comment]
P.S., I would love to get this merged because my plan is to then move on to other categories of improvements such as surnames with spaces (e.g., Van Damm and St. John) and Korean-American names. I know you are busy, but I still want a way to work with your useful project.

**Comment by fgregg** [comment]
Hi @az0, sorry for the slow response. Jean is not longer maintaining this project.  I would recommend adding more training data!

**Comment by az0** [comment]
@fgregg  OK, all tests are now passing.

File: tests/test_data_labeled.xml (hunk 12 above)
Comment (fgregg) [reply tests/test_data_labeled.xml]: @az0, thanks for picking this back up again. Did you address this?

Comment (az0) [reply tests/test_data_labeled.xml]: @fgregg Yes, I removed STROTHEIDE from the training set in this commit https://github.com/az0/probablepeople/commit/4e88092462e296f51cbfdfee6dc0256836b373ad

**Pull request merged**